stdlib/public/core/MemoryLayout.swift

```diff
@@ -0,0 +1,82 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Accesses the memory layout of `T` through its
+/// `size`, `stride`, and `alignment` properties
+public struct MemoryLayout<T> {
+
+  /// Returns the contiguous memory footprint of `T`.
+  ///
+  /// Does not include any dynamically-allocated or "remote" 
+  /// storage. In particular, `MemoryLayout<T>.size`, when 
+  /// `T` is a class type, is the same regardless of how many 
+  /// stored properties `T` has.
+  @_transparent
+  public static var size: Int { return _sizeof(T.self) }
+
+  /// For instances of `T` in an `Array<T>`, returns the number of
+  /// bytes from the start of one instance to the start of the
+  /// next. This is the same as the number of bytes moved when an
+  /// `UnsafePointer<T>` is incremented. `T` may have a lower minimal
+  /// alignment that trades runtime performance for space
+  /// efficiency. The result is always positive.
+  @_transparent
+  public static var stride: Int { return _strideof(T.self) }
+
+  /// Returns the default memory alignment of `T`.
+  @_transparent
+  public static var alignment: Int { return _alignof(T.self) }
+}
+
+extension MemoryLayout {
+  @_transparent
+  init(_ : @autoclosure () -> T) {}
+
+  @_transparent
+  public static func of(_ candidate : @autoclosure () -> T) -> MemoryLayout<T>.Type {
+    return MemoryLayout.init(candidate).dynamicType
+  }
+}
```

Comment:
I just copy pasted it from [the proposal](https://github.com/apple/swift-evolution/blob/master/proposals/0101-standardizing-sizeof-naming.md). 😓 
I'll test it.